pkg/registry/generic/matcher.go

```diff
@@ -79,6 +84,21 @@ func (s *SelectionPredicate) MatchesSingle() (string, bool) {
 	return "", false
 }
 
+// Index will return (value, true) if and only if s.Field matches on the object's
+// field define by the index name.
+func (s *SelectionPredicate) Index(indexName string) (string, bool) {
+	if s.IndexField == nil {
+		return "", false
+	}
+	indexField, supported := s.IndexField(indexName)
```

Comment:
I think we can consider a pattern of:
"field.<field_name>"
"label.<label_name>"
However, so far I only added a TODO (I think we don't need it).